### PR TITLE
Journal at the end of the universe

### PR DESCRIPTION
**Pull request by pleroy**
Complete with some testing.  The recording should be fully working.  Quite a bit left to do for replaying.

### CODE REVIEW

File: ksp_plugin_adapter/interface.cs [hunk 1]
```diff
@@ -5,45 +5,45 @@
 namespace principia {
 namespace ksp_plugin_adapter {
 
-[StructLayout(LayoutKind.Sequential)]
+  [StructLayout(LayoutKind.Sequential)]
```
Comment (eggrobin) [inline ksp_plugin_adapter/interface.cs]: The merge messed with indent everywhere in this file.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 2]
```diff
@@ -193,8 +193,8 @@ where is_in_physics_bubble(vessel)
 
   private void UpdateBody(CelestialBody body, double universal_time) {
     plugin_.UpdateCelestialHierarchy(
-        body.flightGlobalsIndex,
-        body.orbit.referenceBody.flightGlobalsIndex);
+                             body.flightGlobalsIndex,
```
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Same here.

File: ksp_plugin_adapter/interface.cs (hunk 1 above)
Comment (pleroy) [inline ksp_plugin_adapter/interface.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 2 above)
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

**Pull request merged**